### PR TITLE
support more than one matcher for Hash insert

### PR DESCRIPTION
**Pull request by Serjick**

### CODE REVIEW

File: src/Utility/Hash.php [hunk 1]
```diff
@@ -300,12 +300,10 @@ public static function insert(array $data, $path, $values = null)
...
-                if ($conditions && static::_matches($v, $conditions)) {
-                    $data[$k] = array_merge($v, $values);
-                    continue;
-                }
-                if (!$conditions) {
-                    $data[$k] = static::insert($v, $nextPath, $values);
+                if (!$conditions || static::_matches($v, $conditions)) {
+                    $data[$k] = $nextPath
+                        ? static::insert($v, $nextPath, $values)
+                        : array_merge($v, (array) $values);
```
Comment (markstory) [inline src/Utility/Hash.php]: This causes a [PHPCS error](https://travis-ci.org/cakephp/cakephp/jobs/71939961).

File: tests/TestCase/Utility/HashTest.php [hunk 2]
```diff
@@ -1476,6 +1476,17 @@ public function testInsertMulti()
             4 => ['Item' => ['id' => 5, 'title' => 'fifth']],
         ];
         $this->assertEquals($expected, $result);
+
+        $data[3]['testable'] = true;
+        $result = Hash::insert($data, '{n}[testable].Item[id=/\b2|\b4/].test', 2);
```
Comment (markstory) [inline tests/TestCase/Utility/HashTest.php]: My own qualm with this is that this 'feature' is not implemented in other methods that use the hash path syntax. This introduces inconsistencies that make using Hash harder as the syntax harder to use as people have to remember which selectors are supported where.

Comment (Serjick) [inline tests/TestCase/Utility/HashTest.php]: Path syntax already inconsistent for Hash extract and insert methods. I can specify path as "{n}[testable].Item[id=/\b2|\b4/].id" for extract, but not for insert. Don't know anything about other methods, never used them.

File: src/Utility/Hash.php (hunk 1 above)
Comment (Serjick) [inline src/Utility/Hash.php]: fixed

File: tests/TestCase/Utility/HashTest.php (hunk 2 above)
Comment (markstory) [inline tests/TestCase/Utility/HashTest.php]: Fair point, we should also implement these additions to remove(). I think that is the only other method that needs changing at this point.

**Comment by lorenzo** [comment]
This looks good to me... it should be added to the docs :+1:

**Comment by markstory** [comment]
@lorenzo I'll merge and update the docs tonight.

**Pull request merged**